### ISSUE TITLE
SQL DDL error `InvalidColumnNameException["_foo" conflicts with system column pattern]`

### ISSUE THREAD

**Issue by amotl** [crate/crate#15161]
### CrateDB version

Any.

### CrateDB setup information

Docker standard.

### Problem description

SQL DDL definitions using column names with leading underscore characters fail.
```sql
CREATE TABLE testdrive ("_foo" INT);
```
```
InvalidColumnNameException["_foo" conflicts with system column pattern]
```

<details><summary>Details</summary>
<p>

Surprisingly, it works when appending a numeric character, or when using uppercase letters.
```sql
CREATE TABLE testdrive ("_foo2" INT);
```
```sql
CREATE TABLE testdrive2 ("_FOO" INT);
```
```
CREATE OK, 1 row affected (0.265 sec)
```

</p>
</details> 

### Steps to Reproduce

See above.

### Actual Result

Fails.

### Expected Result

Works.

**Comment by matriv** [comment]
Not sure if we want to change this pattern validation for column names...

**Comment by amotl** [comment]
### Background

Meltano maintains a few columns to keep track of record metadata. An example schema is displayed below.

- Docs: https://sdk.meltano.com/en/latest/implementation/record_metadata.html
- Code: https://github.com/meltano/sdk/blob/v0.34.0/singer_sdk/sinks/core.py#L235-L319

### Example DDL
```sql
CREATE TABLE IF NOT EXISTS "melty"."test_carts" (
   "id" BIGINT NOT NULL,
   "fruits" ARRAY(OBJECT(DYNAMIC)),
   "_sdc_extracted_at" TIMESTAMP WITHOUT TIME ZONE,
   "_sdc_received_at" TIMESTAMP WITHOUT TIME ZONE,
   "_sdc_batched_at" TIMESTAMP WITHOUT TIME ZONE,  
   "_sdc_deleted_at" TIMESTAMP WITHOUT TIME ZONE,
   "_sdc_sequence" BIGINT,
   "_sdc_table_version" BIGINT,
   "_sdc_sync_started_at" BIGINT,
   PRIMARY KEY ("id")
)
```

### Thoughts
It would be so sweet if that would not need a workaround on the dialect level, but it is not a blocker, because a workaround can be made. However, it would be more standards-compliant, and while I see CrateDB reserves `_id` already, and maybe others, we can see on behalf of this example is that it is harsh to block the whole namespace.

**Comment by amotl** [comment]
All right. I used double underscores `__` as a table prefix now, to designate "private/hidden" tables, see [sinks.py#L40-L50](https://github.com/crate-workbench/meltano-target-cratedb/blob/e10e76f03597db87ce7a7c5044075ffa13c8a374/target_cratedb/sinks.py#L40-L50).

**Comment by amotl** [comment]
Yes, using `_`-prefixed columns is a common pattern to introduce a "private namespace", in the same way like programming languages are doing it. In both cases, it is the programmer or system designer who chooses what is private or not. In this spirit, many advanced data integration frameworks are following that pattern, and use `_`-prefixed column names to declare them "private", i.e. they are not connected to the core domain the database table is about, but usually include framework-specific metadata information instead.

As long as CrateDB is blocking that namespace, we are pretty much stranded to achieve integrations with those frameworks, while other databases can do it easily. As mentioned above, workarounds are mostly not viable or very limited.

- https://github.com/crate-workbench/meltano-target-cratedb/issues/53
- https://github.com/crate/dlt-cratedb/issues/3 ([see also](https://github.com/crate/crate-clients-tools/issues/86#issuecomment-2466848465))
- https://github.com/crate/cratedb-fivetran-destination/issues/5
- https://github.com/crate/cratedb-estuary/issues/13

**Comment by kneth** [comment]
To be able to leverage stock Postgres tools, we must be able to support the same table names as PostgreSQL. From [PostgreSQL 18 documentation](https://www.postgresql.org/docs/18/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS):

> SQL identifiers and key words must begin with a letter (a-z, but also letters with diacritical marks and non-Latin letters) or an underscore (_). Subsequent characters in an identifier or key word can be letters, underscores, digits (0-9), or dollar signs ($).

**Comment by mfussenegger** [comment]
Lifting the restriction in 6.2.0 with https://github.com/crate/crate/pull/18551
